File: solutions/0345-reverse-vowels-of-a-string/solution.java
class Solution {
    public String reverseVowels(String s) {
        char[] str=s.toCharArray();
        char[] ss=new char[str.length];int k=0;
        for(int i=0;i<str.length;i++)
        {
            if(str[i]=='A'||str[i]=='E'||str[i]=='I'||str[i]=='O'||str[i]=='U'||str[i]=='a'||str[i]=='e'||str[i]=='i'||str[i]=='o'||str[i]=='u')
            {
                ss[k++]=str[i];
            }
        }
        k=0;
        for(int i=str.length-1;i>=0;i--)
        {
            if(str[i]=='A'||str[i]=='E'||str[i]=='I'||str[i]=='O'||str[i]=='U'||str[i]=='a'||str[i]=='e'||str[i]=='i'||str[i]=='o'||str[i]=='u')
            {
                str[i]=ss[k++];
            }
        }
        return new String(str);
    }
}

File: solutions/0345-reverse-vowels-of-a-string/solution.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        ls=[i for i in s if i in "aeiouAEIOU"][::-1]
        n=list(s)
        k=0
        for i in range(len(n)):
            if n[i] in "AEIOUaeiou":
                n[i]=ls[k]
                k+=1
        return "".join(n)
